File: 8_kyu/Switch_it_Up.jl
"""
  Returns english names of integers 0,1,..,9.
  Returns decimal number for other numbers.
"""
function switchitup(number::Real)::String
  isinteger(number) && 0 ≤ number < length(ENGLISH_NUMBER_NAMES) ? ENGLISH_NUMBER_NAMES[number+1] : string(number)
end

const ENGLISH_NUMBER_NAMES = ("Zero", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine")

File: 8_kyu/Switch_it_Up.r
# Returns name of number value.
# Returns decimal number als fallback.
switch_it_up <- function(number){
  if( (number+1) %in% seq_along(number_names) ) 
    number_names[number+1]
  else
    as.character(number)
}

# Name of number index-1. 
number_names <- c("Zero","One","Two","Three","Four","Five","Six","Seven","Eight","Nine")
